engine/class_modules/sc_death_knight.cpp

```diff
@@ -6235,22 +6235,35 @@ struct scourge_strike_t : public scourge_strike_base_t
 
 // Soul Reaper ==============================================================
 
+struct soul_reaper_execute_t : public death_knight_spell_t
+{
+  soul_reaper_execute_t( death_knight_t* p, const std::string& options_str ) :
+    death_knight_spell_t( "soul_reaper_execute", p, p -> spell.soul_reaper_execute )
+  {
+    parse_options( options_str );
```

Comment:
Parsing options is only necessary for APL actions, you don't need it here (and you should set background to true)